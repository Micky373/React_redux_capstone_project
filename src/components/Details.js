import React from 'react';
import { useSelector } from 'react-redux';
import { useParams, Link } from 'react-router-dom';

const Details = () => {
  const params = useParams();
  const { country } = params;
  const countries = useSelector((state) => state);
  const covidData = countries.data.filter((data) => data.CountryName === country);
  const countryFlag = `https://countryflagsapi.com/png/${country}`;

  return (
    <>
      <div className="nav_bar">
        <Link to="/"><i className="fa fa-angle-left nav_buttons" aria-hidden="true" /></Link>
        <span className="nav_header">COVID cases per country</span>
        <i className="fa fa-microphone microphone_button" aria-hidden="true" />
        <i className="fa fa-gear nav_buttons" aria-hidden="true" />
      </div>
      <div className="details_container">
        <div>
          <h1 className="description_header">{`${country}'s Covid Data`}</h1>
        </div>
        <img className="country_flag" src={countryFlag} alt="country flag" />
        <div>
          {countries !== undefined
            ? (
              <div className="description_container">
                <span className="description">
                  Country:&nbsp;&nbsp;
                  {covidData[0].CountryName}
                </span>
                <span className="description">
                  Date:&nbsp;&nbsp;
                  {covidData[0].Date}
                </span>
                <span className="description">
                  Total Confirmed Cases:&nbsp;&nbsp;
                  {covidData[0].TotalConfirmed}
                </span>
                <span className="description">
                  New Confirmed Cases:&nbsp;&nbsp;
                  {covidData[0].NewConfirmed}
                </span>
                <span className="description">
                  New Recovered:&nbsp;&nbsp;
                  {covidData[0].NewRecovered}
                </span>
                <span className="description">
                  Total Deaths:&nbsp;&nbsp;
                  {covidData[0].TotalDeath}
                </span>
                <span className="description">
                  New Deaths:&nbsp;&nbsp;
                  {covidData[0].NewDeath}
                </span>
              </div>
            ) : <h1>***Loading***</h1>}
        </div>
      </div>
    </>
  );
};

export default Details;
