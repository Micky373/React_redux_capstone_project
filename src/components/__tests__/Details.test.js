import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { render, screen } from '@testing-library/react';
import Details from '../Details';
import store from '../../redux/configureStore';
import { HashRouter } from 'react-router-dom';
import '@testing-library/jest-dom';

jest.mock('../../redux/covidData/covidData.js');
const element = (
  <Provider store={store}>
    <HashRouter>
      <Details />
    </HashRouter>
  </Provider>
);

it('it works', () => {
  const tree = renderer.create(element).toJSON();
  expect(tree).toMatchSnapshot();
});

it('Render check', () => {
  render(element);
  const dataRender = screen.getByText('COVID cases per country');
  expect(dataRender).toBeInTheDocument();
});
