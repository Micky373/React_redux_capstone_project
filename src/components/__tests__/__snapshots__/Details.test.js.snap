// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it works 1`] = `
Array [
  <div
    className="nav_bar"
  >
    <a
      href="#/"
      onClick={[Function]}
    >
      <i
        aria-hidden="true"
        className="fa fa-angle-left nav_buttons"
      />
    </a>
    <span
      className="nav_header"
    >
      COVID cases per country
    </span>
    <i
      aria-hidden="true"
      className="fa fa-microphone microphone_button"
    />
    <i
      aria-hidden="true"
      className="fa fa-gear nav_buttons"
    />
  </div>,
  <div
    className="details_container"
  >
    <div>
      <h1
        className="description_header"
      >
        undefined's Covid Data
      </h1>
    </div>
    <img
      alt="country flag"
      className="country_flag"
      src="https://countryflagsapi.com/png/undefined"
    />
    <div>
      <h1>
        ***Loading***
      </h1>
    </div>
  </div>,
]
`;
